lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
It's taken me a while to get my head around this, but I think I understand your proposal. Can I try to outline it to get a clearer understanding?

- GET all widgets: `/wp/v2/widgets`
- GET a specific widget: `/wp/v2/widgets/:id`
- GET with `OPTIONS` request for schema of a specific widget: `/wp/v2/widgets/:id`
- GET all widgets of a specific type: `/wp/v2/widgets/filter[widget_type]`
- GET with `OPTIONS` request for schema of a specific type: `/wp/v2/widgets/:type`
- POST to create a widget (type defined in post data): `/wp/v2/widgets`
- PATCH to create a widget: `/wp/v2/widgets/:id`
- DELETE to delete a widget: `/wp/v2/widgets/:id`

Is that correct?

The `/wp/v2/widgets/:id` and `/wp/v2/widgets/:type` clash feels a bit dirty to me, to be honest. I can see someone doing, for instance, a "Better Recent Posts" widget and giving it a type of `recent-posts-2`. But I don't see any way around it short of creating a new endpoint, `/wp/v2/widget-types`, entirely for the `OPTIONS` request.

(Also, #18 has some code cleanup that's worth looking at if this PR is eventually merged.)